From 3ddeeedb742abb1941ed5e7fdc87ad5c6623f257 Mon Sep 17 00:00:00 2001
From: Vishwath Mohan <vishwath@google.com>
Date: Tue, 2 May 2017 12:08:58 -0700
Subject: [PATCH 06/13] Remove hardcoded LOCAL_MODULE_PATHS from vendor
 components. (bt)

This CL replaces all LOCAL_MODULE_PATH_* definitions for vendor shared
libraries with LOCAL_VENDOR_MODULE := true instead. This appropriately
generates sanitized versions of these libraries to /data/asan/* while
generating the stock versions in /vendor/lib* as desired.

Needed for ASAN builds to work correctly.

Bug: 37740897
Test: m -j40 && SANITIZE_TARGET="address" m -j40 # shared libs in
$ANDROID_PRODUCT_OUT/vendor/lib* do not contain asan symbols, while
the ones in $ANDROID_PRODUCT_OUT/data/asan/vendor/lib* do.

Change-Id: Iaa67166fa32f236313cef111c43bc927076d64fa
---
 libbt-vendor/Android.mk | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/libbt-vendor/Android.mk b/libbt-vendor/Android.mk
index 23eced9..cb66f5a 100644
--- a/libbt-vendor/Android.mk
+++ b/libbt-vendor/Android.mk
@@ -70,12 +70,7 @@ LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE_CLASS := SHARED_LIBRARIES
 LOCAL_MODULE_OWNER := qcom
 
-ifdef TARGET_2ND_ARCH
-LOCAL_MODULE_PATH_32 := $(TARGET_OUT_VENDOR)/lib
-LOCAL_MODULE_PATH_64 := $(TARGET_OUT_VENDOR)/lib64
-else
-LOCAL_MODULE_PATH := $(TARGET_OUT_VENDOR_SHARED_LIBRARIES)
-endif
+LOCAL_VENDOR_MODULE := true
 
 ifeq ($(QCOM_BT_USE_BTNV),true)
 LOCAL_CFLAGS += -DBT_NV_SUPPORT
-- 
2.17.1

